docs/theming.md

```diff
@@ -2,37 +2,92 @@
 title: 'Theming'
 ---
 
-Theming is provided by the library [`emotion-theming`](https://emotion.sh/docs/emotion-theming).
+Emotion 10 and above doesn’t have any explicit theming API, instead, we recommend using React’s Context and Hooks which work well together with Emotion. This has many benefits over having a built in theming API which are explained below.
 
-```bash
-npm install -S emotion-theming
+We recommend creating a file which exports a ThemeProvider component and a useTheme hook.
+
+```jsx
+import { createContext, useContext } from 'react'
+
+let defaultTheme = {
+  colors: {
+    primary: 'hotpink',
+    secondary: 'green'
+  }
+}
+
+let ThemeContext = createContext(defaultTheme)
+
+export let ThemeProvider = ThemeContext.Provider
+
+export let useTheme = () => useContext(ThemeContext)
 ```
 
-Add `ThemeProvider` to the top level of your app and access the theme with `props.theme` in a styled component or provide a function that accepts the theme as the css prop. The api is laid out in detail [in the documentation](https://emotion.sh/docs/emotion-theming).
+The theme can be used with css prop like this.
 
 ```jsx
-// @live
 /** @jsx jsx */
 import { jsx } from '@emotion/core'
+import { useTheme } from './theme'
+
+let Button = props => {
```

Comment:
this recommendation suffers from `shouldForwardProp` problem, which really gets annoying long term from DX perspective

Reply:
Maybe this example shouldnt mention `button` but rather a composite component

Reply:
right, ive misjudged the example by not thinking about it too much and having styled API with interpolations in mind, css prop doesnt suffer from this

Reply:
We should probably dedicate a page on the docs dedicated to mapping solutions to "known" apis like `withComponent`, `shouldForwardProp`, etc. We should also show how easy it is to add further functionality to components this way.